salt/client/__init__.py

```diff
@@ -951,7 +951,7 @@ def cmd_iter_no_block(
 
                 self._clean_up_subscriptions(pub_data["jid"])
         finally:
-            if not was_listening:
+            if not was_listening and not self.event.pending_events:
```

Comment:
Hmm, I'm not happy with leaving the event bus unclosed. We've had memory leaks because of that.
Waiting until there are no pending events won't work?